core/iwasm/libraries/libc-wasi/libc_wasi_wrapper.c

```diff
@@ -961,8 +961,9 @@ get_timeout_for_poll_oneoff(const wasi_subscription_t *in,
     for (i = 0; i < nsubscriptions; ++i) {
         const __wasi_subscription_t *s = &in[i];
         if (s->u.type == __WASI_EVENTTYPE_CLOCK
-            && (s->u.u.clock.flags & __WASI_SUBSCRIPTION_CLOCK_ABSTIME) == 0) {
-            timeout = min(timeout, s->u.u.clock.timeout);
+            && (s->u.u.clock.flags & __WASI_SUBSCRIPTION_CLOCK_ABSTIME) == 0
+            && s->u.u.clock.timeout < timeout) {
+            timeout = s->u.u.clock.timeout;
```

Comment:
Good catch; I wonder if other use of min (https://github.com/bytecodealliance/wasm-micro-runtime/pull/2152/files#diff-18a1c12794da572224835cdd9491ff7f93bc7ff81249c75ead3052e7f815f9f8R1021) would potentially be invalid too as it can go over `size_t` type?

Reply:
From my point of view, looks like the answer is "no" 

It's defined in the same file and defined to compare two `size_t` which is not the safest way of doing that 

Probably we can refactor it to compare two `uint64_t`/`int64_t` . It'll be much safer I think because cast to a  wider type is always ok

Reply:
Fair concern, fixed